service/src/LevelZeroImp.hpp

```diff
@@ -114,9 +128,25 @@ namespace geopm
                 std::vector<zes_pwr_handle_t> power_domain;
                 mutable std::vector<uint64_t> cached_energy_timestamp;
 
+                //ZE Context used for ZET data collection
+                std::vector<ze_context_handle_t> context;
+
+                // required for L0 metric querying
+                std::vector<uint32_t> num_metric;
+                std::vector<uint32_t> num_reports;
+                std::vector<bool> metric_domain_cached;
+                std::vector<ze_event_pool_handle_t> event_pool;
+                std::vector<ze_event_handle_t> event; //TODO: rename metric_notification_event?
+                std::vector<zet_metric_streamer_handle_t> metric_streamer;
+                std::vector<zet_metric_group_handle_t> metric_group_handle; //compute basic only
+
+                // required for L0 metric result tracking
+                mutable std::vector<std::map<std::string, std::vector<double>>> m_metric_data;
```

Comment:
Q: why does this need the mutable specifier?

Reply:
It doesn't anymore!